### PR TITLE
Added decimal writing for CSV writer

### PR DESCRIPTION
**Pull request by kaatish**
Addresses #7110 

column_to_strings_fn was specialized for fixed point type to enable support for csv writer. A test was added to validate output file created by csv writer for decimal type column.

### CODE REVIEW

File: cpp/tests/io/csv_test.cpp [hunk 1]
```diff
@@ -307,6 +319,42 @@ TYPED_TEST(CsvReaderNumericTypeTest, SingleColumn)
...
+  std::ifstream read_result_file(filepath);
+  assert(read_result_file.is_open());
+
+  std::copy(std::istream_iterator<std::string>(read_result_file),
+            std::istream_iterator<std::string>(),
+            std::back_inserter(result_strings));
+
+  read_result_file.close();
+
+  ASSERT_EQ(result_strings, reference_strings);
```
Comment (davidwendt) [inline cpp/tests/io/csv_test.cpp:355]:
```suggestion
  EXPECT_EQ(result_strings, reference_strings);
```
There is no code after this line to protect.

**Review by ttnghia — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

File: cpp/tests/io/csv_test.cpp [hunk 2]
```diff
@@ -307,6 +319,42 @@ TYPED_TEST(CsvReaderNumericTypeTest, SingleColumn)
...
+{
+  std::vector<std::string> reference_strings = {
+    "1.23", "-8.76", "5.43", "-0.12", "0.25", "-0.23", "-0.27", "0.00", "0.00"};
+
+  cudf::test::strings_column_wrapper strings(reference_strings.begin(), reference_strings.end());
+
+  using DecimalType = TypeParam;
+  auto input_column = cudf::strings::to_fixed_point(
+    cudf::strings_column_view(strings),
+    cudf::data_type{cudf::type_to_id<DecimalType>(), numeric::scale_type{-2}});
```
Comment (vuule) [inline cpp/tests/io/csv_test.cpp:332]: We should include a test with positive scale.

File: cpp/tests/io/csv_test.cpp [hunk 3]
```diff
@@ -307,6 +319,42 @@ TYPED_TEST(CsvReaderNumericTypeTest, SingleColumn)
   expect_column_data_equal(std::vector<TypeParam>(sequence, sequence + num_rows), view.column(0));
 }
 
+TYPED_TEST(CsvFixedPointWriterTest, SingleColumn)
+{
+  std::vector<std::string> reference_strings = {
+    "1.23", "-8.76", "5.43", "-0.12", "0.25", "-0.23", "-0.27", "0.00", "0.00"};
```
Comment (vuule) [inline cpp/tests/io/csv_test.cpp:325]: We should also include null elements in the test.

**Review by vuule — APPROVED** [review]
Looks good :)

**Review by devavret — APPROVED** [review]

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**